### PR TITLE
Fix client-go fake client example flake

### PR DESCRIPTION
**Pull request by tnozicka**
**What type of PR is this?**
/kind flake

**What this PR does / why we need it**:
The test is flaky.  Seen e.g. in https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/batch/pull-ci-openshift-origin-master-unit/3803#githubcomopenshiftoriginvendork8sioclient-goexamplesfake-client-testfakeclient (according to @smarterclayton with 1/8 frequency) reproduced locally with pure kube:
```
go test -race ./vendor/k8s.io/client-go/examples/fake-client -run='TestFakeClient' -failfast -count=10000 -parallel=1000
--- FAIL: TestFakeClient (5.00s)
    main_test.go:75: Informer did not get the added pod
FAIL
FAIL	k8s.io/kubernetes/vendor/k8s.io/client-go/examples/fake-client	69.991s
```

**Special notes for your reviewer**:
I think the main issue was setting the timeout before spinning up the informers so locking and load would cause it to timeout.

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

/cc @smarterclayton @kubernetes/sig-api-machinery-pr-reviews

### CODE REVIEW

File: staging/src/k8s.io/client-go/examples/fake-client/main_test.go [hunk 1]
```diff
@@ -31,7 +31,8 @@ import (
 // TestFakeClient demonstrates how to use a fake client with SharedInformerFactory in tests.
 func TestFakeClient(t *testing.T) {
 	// Use a timeout to keep the test from hanging.
-	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	ctx, cancel := context.WithCancel(context.Background())
```
Comment (sttts) [inline staging/src/k8s.io/client-go/examples/fake-client/main_test.go]: wait.ForeverTestTimeout

File: staging/src/k8s.io/client-go/examples/fake-client/main_test.go [hunk 2]
```diff
@@ -55,23 +55,19 @@ func TestFakeClient(t *testing.T) {
...
+		t.Fatalf("error injecting pod add: %v", err)
 	}
 
-	// Wait and check result.
-	<-ctx.Done()
 	select {
 	case pod := <-pods:
 		t.Logf("Got pod from channel: %s/%s", pod.Namespace, pod.Name)
-	default:
+	case <-time.After(15 * time.Second):
```
Comment (sttts) [inline staging/src/k8s.io/client-go/examples/fake-client/main_test.go]: wait.ForeverTestTimeout

File: staging/src/k8s.io/client-go/examples/fake-client/main_test.go (hunk 1 above)
Comment (tnozicka) [reply staging/src/k8s.io/client-go/examples/fake-client/main_test.go]: this is not using timeout, just the function scope

**Pull request merged**